### ISSUE TITLE
Formatter for Input (not issue #11198)

### ISSUE THREAD

**Issue by WolfeLogic-LHR** [ant-design/ant-design#20741]
- [ ✓] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?

The display of user-input updating dynamically for visual validation and improved user experience. The three examples that immediately come to mind are:
- Phone numbers | '13263594569' versus '1 (326) 359 - 4569' 
- Credit card numbers | '4147362545691247' versus '4147 3625 4569 1247'
- Personal identification numbers, in the USA, Social Security | '594689465' versus '594 68 9465'

We can always regex the input prior to firing off our own internal API calls. This issue [does **not** relate to input validation](https://ant.design/components/form/#Validation-Rules).

Enabling a user to visually confirm that s/he has input all 12 digits of their credit card or the 15 digits of their far-less confidential library book, or the GPS coordinates for a geocaching site, or... There are numerous examples that include numerical values but must be typed as strings. To my knowledge, these issues do not fall under the existing `normalize` and `validate` utilities. It would be tremendously useful to be able to format input as a user types into the field.

We should be able to perform the feature [illustrated in this JSFiddle](https://jsfiddle.net/rafj3md0/) through use of a `formatter` and `parser` utilities on the ant design `<Input />` feature

### What does the proposed API look like?
Use the existing `formatter` and `parser` from `InputNumber` without the inherent HTML number input step control

```
.ant-input-number-handler-wrap {
     display: none;
     }
```

**Comment by WolfeLogic-LHR** [comment]
![image](https://user-images.githubusercontent.com/42189943/72169838-155c5780-338d-11ea-9e09-02c5012044ba.png)
This is a woefully substandard and inferior response to what is clearly a known issue. All I want to know is that it will be addressed in Ant Design 4.0

**Comment by zombieJ** [comment]
Format input should have a lot of edge case. We recommend to use other library to handle this. We recommend `text-mask` before but it failed after react new life cycle update. Please feel free to add recommend lib with us that will add in recommend page.

ref: https://github.com/ant-design/ant-design/issues/17104
